src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDK.java

```diff
@@ -0,0 +1,133 @@
+package cloud.fogbow.ras.core.plugins.interoperability.azure.publicip.sdk;
+
+import java.util.concurrent.ExecutorService;
+
+import org.apache.log4j.Logger;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.network.NetworkInterface;
+import com.microsoft.azure.management.network.NetworkSecurityGroup;
+import com.microsoft.azure.management.network.PublicIPAddress;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+
+import cloud.fogbow.ras.constants.Messages;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.util.AzureSchedulerManager;
+import rx.Completable;
+import rx.Observable;
+import rx.Scheduler;
+import rx.schedulers.Schedulers;
+
+public class AzurePublicIPAddressOperationSDK {
+    
+    @VisibleForTesting
+    static final Logger LOGGER = Logger.getLogger(AzurePublicIPAddressOperationSDK.class);
+    @VisibleForTesting
+    static final int FULL_CAPACITY = 2;
+
+    private final String resourceGroupName;
+
+    private Scheduler scheduler;
+
+    public AzurePublicIPAddressOperationSDK(String resourceGroupName) {
+        ExecutorService executor = AzureSchedulerManager.getPublicIPAddressExecutor();
+        this.scheduler = Schedulers.from(executor);
+        this.resourceGroupName = resourceGroupName;
+    }
+
+    public void subscribeAssociatePublicIPAddress(Azure azure, String instanceId,
+            Observable<NetworkInterface> observable) {
+        
+        setAssociatePublicIPAddressBehaviour(azure, instanceId, observable)
+            .subscribeOn(this.scheduler)
+            .subscribe();
+    }
+
+    @VisibleForTesting
+    Observable<NetworkInterface> setAssociatePublicIPAddressBehaviour(Azure azure, String instanceId,
+            Observable<NetworkInterface> observable) {
+
+        return observable.doOnNext(nic -> {
+            LOGGER.info(Messages.Info.FIRST_STEP_CREATE_PUBLIC_IP_ASYNC_BEHAVIOUR);
+            handleSecurityIssues(azure, instanceId, nic);
```

Comment:
I would use another method name. This is an important step in our operation, therefore we should make it explicit.
We might use associateSecurityGroup. What do you think?
Or you might use the same way of other plugins. Something like: [Link](https://github.com/fogbow/resource-allocation-service/blob/develop/src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureVirtualNetworkOperationSDK.java#L70)

Reply:
Ok, I'll rename this method.